### PR TITLE
feature(artifacts): add tests and pipelines for testing rocky9

### PR DESCRIPTION
**Pull request by fruch**
Rocky Linux 9 was recently released and we want to start testing it

Ref: https://rockylinux.org/news/rocky-linux-9-0-ga-release/
Ref: https://gcloud-compute.com/images.html

## Testing

- [x] - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/fruch/job/artifacts-rocky9-test/8/
- [x] - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/fruch/job/artifacts-rocky9-test/12/

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by fruch** [comment]
this currently would fail cause not available image:
```
google.api_core.exceptions.NotFound: 404 GET https://compute.googleapis.com/compute/v1/projects/rocky-linux-cloud/global/images/family/rocky-linux-9: 
The resource 'projects/rocky-linux-cloud/global/images/family/rocky-linux-9' was not found
```

**Review by fgelcer — APPROVED** [review]
LGTM

**Comment by roydahan** [comment]
@fruch where did we get this requirement from?
Is this still relevant?

**Comment by fruch** [comment]
> @fruch where did we get this requirement from?
> Is this still relevant?

From @avikivity 

This led to start pushing to repackage cqlsh

I've used this PR to verify the installation of the new cqlsh

This is pending the merge of cqlsh repackaging

File: sdcm/mgmt/common.py [hunk 1]
```diff
@@ -25,6 +25,7 @@ def get_distro_name(distro_object):
         Distro.OEL7: "centos7",
         Distro.OEL8: "centos8",
         Distro.ROCKY8: "centos8",
+        Distro.ROCKY9: "centos8",
```
Comment (fgelcer) [inline sdcm/mgmt/common.py:28]: shouldn't it be `centos9` ?

Comment (fruch) [reply sdcm/mgmt/common.py:28]: It's just a mapping to get the manger repo

centos8 is just a key pointing to the same repo as centos7

**Review by fgelcer — APPROVED** [review]
LGTM

File: jenkins-pipelines/nonroot-offline-install/artifacts-rocky9.jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#! groovy
+
+// trick from https://github.com/jenkinsci/workflow-cps-global-lib-plugin/pull/43
+def lib = library identifier: 'sct@snapshot', retriever: legacySCM(scm)
+
+artifactsPipeline(
+    test_config: 'test-cases/artifacts/rocky8.yaml',
```
Comment (soyacz) [inline jenkins-pipelines/nonroot-offline-install/artifacts-rocky9.jenkinsfile:7]: filename refers to rocky9 while here is rocky8

**Review by soyacz — COMMENTED** [review]
one comment, otherwise looks ok.

File: test-cases/artifacts/rocky9.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+gce_n_local_ssd_disk_db: 1
+instance_provision: "spot"
+instance_provision_fallback_on_demand: true
+logs_transport: 'ssh'
+n_db_nodes: 1
+n_loaders: 0
+n_monitor_nodes: 0
+nemesis_class_name: 'NoOpMonkey'
+scylla_linux_distro: 'centos'
+scylla_repo: 'https://s3.amazonaws.com/downloads.scylladb.com/rpm/centos/scylla-3.3.repo'
```
Comment (vponomaryov) [inline test-cases/artifacts/rocky9.yaml:16]: Are you sure about the `3.3`?

Comment (fruch) [reply test-cases/artifacts/rocky9.yaml:16]: not really... I was copied a long long time ago, I'll remove it.

File: jenkins-pipelines/nonroot-offline-install/artifacts-rocky9.jenkinsfile (hunk 2 above)
Comment (fruch) [reply jenkins-pipelines/nonroot-offline-install/artifacts-rocky9.jenkinsfile:7]: fixing

**Review by soyacz — APPROVED** [review]

**Review by vponomaryov — APPROVED** [review]
LGTM

**Pull request merged**